### ISSUE TITLE
There is no variable named "dependency" - terragrunt 0.55.4

### ISSUE THREAD

**Issue by philslab-ninja** [gruntwork-io/terragrunt#2955]
**Describe the bug**
With terragrunt 0.55.4 I get errors all over the place when using dependency outputs as inputs

**To Reproduce**
Have a dependency, reference an output of that dependency as an input, and probably have the dependency have dependencies also.

```hcl

dependency "eks_cluster" {
  config_path = "../../eks-cluster"
}

dependency "kms_key_2024" {
  config_path = "../../kms/external-pv-key-2024"
}

inputs = {
  provider_url           = dependency.eks_cluster.outputs.cluster_oidc_issuer_url
  cluster_id             = dependency.eks_cluster.outputs.cluster_name
  cluster_endpoint       = dependency.eks_cluster.outputs.cluster_endpoint
  cluster_ca_cert        = dependency.eks_cluster.outputs.cluster_certificate_authority_data
  oidc_provider_arn      = dependency.eks_cluster.outputs.oidc_provider_arn
  kms_keys_for_ebs_csi   = [dependency.kms_key_2024.outputs.arn]
}
```

When planning / applying I get errors from the referenced modules:

```
ERRO[0000] Error: Unknown variable
            
ERRO[0000]   on /some/path/to/infrastructure-live/test/eu-central-1/e2e2/services/eks-01/eks-cluster/terragrunt.hcl line 47: 
ERRO[0000]   47:   kms_key_for_node_ebs = dependency.external-ebs-key-2024.outputs.arn 
ERRO[0000]                                              
ERRO[0000] There is no variable named "dependency".
    
WARN[0000] Error reading partial config  for dependency eks_cluster: /some/path/to/infrastructure-live/test/eu-central-1/e2e2/services/eks-01/eks-cluster/terragrunt.hcl:47,26-36: Unknown variable; There is no variable named "dependency". 
ERRO[0000] Error: Unknown variable
            
ERRO[0000]   on /some/path/to/infrastructure-live/test/eu-central-1/e2e2/services/eks-01/kms/external-pv-key-2024/terragrunt.hcl line 44: 
ERRO[0000]   44:   ciphertrust_aws_kms_id = dependency.thales_kms.outputs.ciphertrust_aws_kms_id 
ERRO[0000]                                              
ERRO[0000] There is no variable named "dependency".
```

In those modules I have dependencies as well, as an example in that `eks-cluster/terragrunt.hcl` line 47:

```hcl
  kms_key_for_node_ebs = dependency.external-ebs-key-2024.outputs.arn
```

where the dependency is declared earlier as:
```hcl
dependency "external-ebs-key-2024" {
  config_path = "../kms/external-ebs-key-2024"
}
```

**Expected behavior**
No errors or warning should be thrown, as with terragrunt 0.55.3 and earlier

**Nice to have**
- [x] Terminal output

**Versions**
- Terragrunt version: 0.55.4
- Terraform version: 1.7.2
- Environment details (Ubuntu 20.04, Windows 10, etc.): macOS 14.3.1

**Additional context**
I am available on the Gruntwork Slack as `@Philipp`

**Comment by denis256** [comment]
It looks like it is an issue with transient dependency outputs
Happens in my test project:
```
https://github.com/denis256/terragrunt-tests/tree/master/issue-2955
```

**Comment by levkohimins** [comment]
Resolved in [v0.55.6](https://github.com/gruntwork-io/terragrunt/releases/tag/v0.55.6) release.
